### ISSUE TITLE
Train

### ISSUE THREAD

**Issue by Zumbalamambo** [adamcasson/c3d#1]
How do I train in my own dataset?

**Comment by Mohamad73** [comment]
Hello,
Thanks for your effort
I also want to train my own dataset? How can I do it?

**Comment by binhhoangtieu** [comment]
To @Zumbalamambo, @Mohamad73 
You can load this weight into your pre-defined model and add last dense layer with softmax function for your classification.
Good luck

**Comment by adamcasson** [comment]
@binhhoangtieu this will work if they want to fine tune the pre-trained weights.

@Zumbalamambo @Mohamad73 
If you want to train from scratch then something like this will work:

```python
model = C3D(weights=None)
model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])
model.fit(data, labels, epochs=100, batch_size=64)
```

where `data` is your training samples with dimension of `(samples, frames, height, width, channels)` and `labels` is a one-hot encoding of your ground truth labels with dimensions `(samples, classes)`.

This would require your whole training data to be loaded into memory though. That may not be possible which in that case you'll have to use a python generator to load your dataset by batches and then use Keras's `model.fit_generator()` function to train the model.
